### PR TITLE
[ARCTIC-1332] Fix a `ScheduledExecutorService` with coolPoolSize = 0 consumes 100% CPU

### LINKED ISSUE #1332: [Bug]: High CPU occupancy of AMS
### What happened?

After start AMS, Trash worker thread occupies 100% CPU.

image

image

### Affects Versions

master/0.4.1

### What engines are you seeing the problem on?

AMS

### How to reproduce

1. Set `conf/config.yaml`, remove `arctic.ams.trash.clean.thread.pool-size`, or set it to `0`
2. restart AMS

### Relevant log output

_No response_

### Anything else

It happens every time.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

fix #1332 

## Brief change log

  - check thread count > 0 for all thread count configurations of AMS
  - support close expire-snapshots/clean-orphan-files/hive-sync

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by wangtaohz** [comment]
After fixed, CPU usage is normal
image
image

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 1]
```diff
@@ -289,6 +298,32 @@ private void validateConfig(Map<String, Object> systemConfig) {
...
+    }
+    
+    private boolean enabled(Map<String, Object> systemConfig, ConfigOption<Boolean> config) {
+      return (boolean) systemConfig.getOrDefault(config.key(), config.defaultValue());
+    }
+
+    private void validateThreadCount(Map<String, Object> systemConfig, ConfigOption<Integer> config) {
+      int threadCount = (int) systemConfig.getOrDefault(config.key(), config.defaultValue());
+      if (threadCount <= 0) {
+        throw new RuntimeException(
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java:324]: Is `IllegalArgumentException` better here?

File: dist/src/main/arctic-bin/conf/config.yaml [hunk 2]
```diff
@@ -3,15 +3,6 @@ ams:
   admin-password: admin
   server-bind-host: "0.0.0.0"
   server-expose-host: "127.0.0.1"
-  refresh-external-catalog-interval: 180000 # 3min
```
Comment (zhoujinsong) [inline dist/src/main/arctic-bin/conf/config.yaml:6]: Change the config in `site/docs/ch/guilds/deployment.md` too.

Comment (wangtaohz) [reply dist/src/main/arctic-bin/conf/config.yaml:6]: Okay, I will change it.

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java (hunk 1 above)
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java]: Yes, I fixed all of them when AMS configurations are illegal.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**